server/odc-migrate/src/main/java/com/oceanbase/odc/migrate/jdbc/common/V42417HistoricalPartitionPlanMigrate.java

```diff
@@ -70,7 +70,9 @@ public class V42417HistoricalPartitionPlanMigrate implements JdbcMigratable {
     @Override
     public void migrate(DataSource dataSource) {
         JdbcTemplate jdbcTemplate = new JdbcTemplate(dataSource);
-        List<PartitionPlanEntityWrapper> partitionPlanWrappers = generatePartitionPlanEntities(jdbcTemplate);
+        List<PartitionPlanEntityWrapper> partitionPlanWrappers = generatePartitionPlanEntities(jdbcTemplate).stream()
+                .filter(w -> CollectionUtils.isNotEmpty(w.getPartitionPlanTableEntityWrappers()))
+                .collect(Collectors.toList());
```

Comment:
What's the impact of filtering out these disabled plans? Will these plans still exist in ODC 4.2.4?

Reply:
> What's the impact of filtering out these disabled plans?

nothing, the way we cancel a partition plan is to startup a new partition plan and this new partition plan will contain no table or contain tables which is all disabled~

we will encounter a npe if we do not filter out these partition plans.

> Will these plans still exist in ODC 4.2.4

no, we will filter out these partition plan cause that these partition plans will do nothing and there is no necessary to migrate them~

Reply:
So if users disabled a partition plan before V4.2.4, he/she cannot see that disabled plan any more after V4.2.4? Is that what we expected?

Reply:
no, he/she will will can see these disabled partition plans but he/she will not see the detail of these partition plan cause that we do not migrate these data. ODC will return empty response when he/she attempt to view these data which is expected~

Reply:
okay, that makes sense to me~